### DISCUSSION TITLE
Allow US choropleth maps to use strings

### DISCUSSION THREAD

**Discussion by mmr-crexi** [evidence-dev/evidence#2367] [General]
**Feature Description**
Choropleth maps right now require numbers per state.  If there were strings, I could append a '%' to indicate percentage, or have a whole string (percentage, numerator, denominator) per state

**Goal of Feature**
Provide more utility for the choropleth map, and more detail for my users.

**Current Solution**
I have explanatory text below and allow them to look at query results

**Alternative Ideas**
Haven't really thought of any

**Examples/Screenshots**
Not without giving away proprietary info...

**Are you interested in contributing this feature?**
If I only had the time

**Comment by mcrascal** [comment]
Thanks @mmr-crexi. 

Here's how we're thinking of this, lmk if any issues: 

1. We need a numeric value to power the colouring in the choropleth. 
1. We can make sure those values respect formatting tags, similar to the rest of the charts. (e.g. value_pct will render as a properly formatted percentage) 
2. We can look at including an additional arbitrary string in the tooltip label, so that you could drop in additional info if you wanted to.

**Comment by mmr-crexi** [comment]
Ah got it-- so it seems that the field is serving double duty, one for scaling of color values and the other for direct display.  I like the idea of string formatting, that will likely make a lot of sense-- having the ability to say what the numerator and denominator are for each state is where I want to be, because a 10% in Wyoming by population will be a very different number than a 10% in California.
